includes/wc-template-functions.php

```diff
@@ -3018,7 +3018,7 @@ function wc_dropdown_variation_attribute_options( $args = array() ) {
 		$product               = $args['product'];
 		$attribute             = $args['attribute'];
 		$name                  = $args['name'] ? $args['name'] : 'attribute_' . sanitize_title( $attribute );
-		$id                    = $args['id'] ? $args['id'] : sanitize_title( $attribute );
+		$id                    = $args['id'] ? $args['id'] : ( ( ! empty( $args['product']->get_id() ) ) ? 'attribute_' . $args['product']->get_id() . '_' . sanitize_title( $attribute ) : sanitize_title( $attribute ) );
```

Comment:
Instead of doing this there, we can generate the id only in `templates/single-product/add-to-cart/variable.php`. that way, our code is much simpler.